### DISCUSSION TITLE
Where's the missing custom_3d_three-d.rs?

### DISCUSSION THREAD

**Discussion by BreakingLead** [emilk/egui#2889] [Q&A] [answered]
From README.md

**Comment by BreakingLead** [comment]
> 
> Shape::Callback
> Examples:
> 
> https://github.com/emilk/egui/blob/master/examples/custom_3d_three-d.rs
> https://github.com/emilk/egui/blob/master/examples/custom_3d_glow.rs
> Shape::Callback will call your code when egui gets painted, to show anything using whatever the background rendering context is. When using [eframe](https://github.com/emilk/egui/tree/master/crates/eframe) this will be [glow](https://github.com/grovesNL/glow). Other integrations will give you other rendering contexts, if they support Shape::Callback at all.

**Reply by BreakingLead** [reply]
https://github.com/emilk/egui/blob/master/examples/custom_3d_three-d.rs 
404 not found

**Reply by Ionizing** [reply]
Maybe it was removed for some reason. The latest commit about it I found is https://github.com/emilk/egui/tree/a8d5a82a7f4b341b32b8d7b9c1cf429338a0ec43/examples/custom_3d_three-d

**Reply by stonerfish** [reply, marked as answer]
Yes, it has been removed.
From git log...

commit 636a39cbe122ce91ea554920ee7aa4475d8f2de7
Author: Emil Ernerfeldt <emil.ernerfeldt@gmail.com>
Date:   Wed Feb 8 17:16:44 2023 +0100

    Update `glow` to 0.12 (#2695)

    * Remove three-d example due to glow incompatibility

**Reply by BreakingLead** [reply]
But readme.md hadn't been updated.
